### ISSUE TITLE
Support deep-merging lint rule configs

### ISSUE THREAD

**Issue by chriskrycho** [checkupjs/checkup#828]
Checkup tasks' defaults are often very good, but sometimes may need to be tweaked for a given users' needs. For example, our app currently uses Ember M3, which makes extensive use of Ember's `ObjectProxy` and therefore requires `get` in many places that code using mainline Ember Data does *not*—so we need to tweak the Octane task definition of the `no-get` rule to understand that we have many uses of `get` which are *not* violations.

From previous discussions, it seems like the best path forward here may be to allow users to supply lint configs which Checkup can then deeply merge with its defaults.

**Comment by scalvert** [comment]
I agree with this. We should effectively allow task configs to supply eslint configs per-rule, merging those together with the inline config we define within a task. I'll try to figure out a nice API for it, so it's less work for the task author.

Thanks, @chriskrycho!
